testConnectivity.py

```diff
@@ -409,75 +436,157 @@ def testConnectivity(args):
-                        phaseRes_idxm1 = phaseRes_idxm0
-                        phase2Write = phase-2
-                    elif (  phaseRes_idxm2 == args.nPhaseScans and
-                            phaseRes_idxm1 == args.nPhaseScans and
-                            phaseRes_idxm0 == args.nPhaseScans): 
-                        phaseRes_idxm3 = phaseRes_idxm2
-                        phaseRes_idxm2 = phaseRes_idxm1
-                        phaseRes_idxm1 = phaseRes_idxm0
-                        phase2Write = phase-1
-                    elif (phaseRes_idxm1 == args.nPhaseScans and phaseRes_idxm0 == args.nPhaseScans): # Last phase and this phase are good
-                        phaseRes_idxm2 = phaseRes_idxm1
-                        phaseRes_idxm1 = phaseRes_idxm0
-                    elif (phaseRes_idxm0 == args.nPhaseScans): # Only this phase is good
-                        phaseRes_idxm1 = phaseRes_idxm0
-                    else: # Reset
-                        phaseRes_idxm0 = 0
-                        phaseRes_idxm1 = 0
-                        phaseRes_idxm2 = 0
-                        phaseRes_idxm3 = 0
-                        phaseRes_idxm4 = 0
+                phaseCounts = np.array([ dict_phaseScanResults[ohN][vfat*GBT_PHASE_RANGE+ph] for ph in range(0,GBT_PHASE_RANGE) ])
+                allBadPhases = np.where(phaseCounts!=args.nPhaseScans)[0]
+                badPhaseCounts = np.delete(allBadPhases,np.where(allBadPhases==15)[0]) ## remove 15 from the list of bad phases
+                phaseSum = 0
+                if len(badPhaseCounts) == 0:
+                    # First try to set the phase from the lookup table
+                    if phaseIsGood(vfatBoard, vfat, vfatBoard.parentOH.vfatGBTPhases):
+                        phase2Write = vfatBoard.parentOH.vfatGBTPhases[vfat]
+                    else:
+                        # Wonder if this could be done with a lambda...probably not
+                        tmpPhase = -1
+                        if (vfatBoard.parentOH.vfatGBTPhases[vfat] + 4) < 15:
+                            tmpPhase = vfatBoard.parentOH.vfatGBTPhases[vfat] + 4
+                        else:
+                            tmpPhase = vfatBoard.parentOH.vfatGBTPhases[vfat] - 4
+                            pass
+
+                        if phaseIsGood(vfatBoard, vfat, tmpPhase):
+                            phase2Write = tmpPhase
+                            pass
                         pass
-                    pass # End loop over phases
+                    
+                    if (not (phase2Write > -1)):
+                        vfats2Replace.append((ohN,vfat))
+                else:
+                    psum = -1
+                    for b in badPhaseCounts:
+                        WINDOW = 4 ## Good phase search window
+                        SHIFT  = 4 ## Phase shift from bad phase to set
+                        frange  = crange(int(badPhase+1),
```

Comment:
missing import on `crange`

Reply:
Also `badPhase` causes a `nameError` I guess this is supposed to be the array index `b`?